integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/SmokeIT.java

```diff
@@ -140,6 +140,17 @@ public void where_boolean() {
 				.hasDocRefHitsAnyOrder( index.typeName(), "1", "2" )
 				.hasTotalHitCount( 2 );
 
+		query = scope.query()
+				.where( f -> f.or(
+								f.match().field( "integer" ).matching( 1 ),
+								f.match().field( "integer" ).matching( 2 )
+						)
+				)
+				.toQuery();
+		assertThatQuery( query )
+				.hasDocRefHitsAnyOrder( index.typeName(), "1", "2" )
+				.hasTotalHitCount( 2 );
+
```

Comment:
You can skip updating this tests (`SmokeIT`), I'll remove it because it's no longer useful: https://github.com/hibernate/hibernate-search/pull/3153